src/zenml/stack_stores/sql_stack_store.py

```diff
@@ -245,6 +245,94 @@ def register_stack_component(
             session.add(new_component)
             session.commit()
 
+    def update_stack_component(
+        self,
+        component: StackComponentWrapper,
+    ) -> None:
+        """Update a stack component.
+
+        Args:
+            component: The new component to update with.
+
+        Raises:
+            KeyError: If no stack component exists with the given name.
+        """
+        with Session(self.engine) as session:
+            existing_components = session.exec(
+                select(ZenStackComponent).where(
+                    ZenStackComponent.component_type == component.type
+                )
+            ).all()
+            if not existing_components:
+                raise StackComponentExistsError(
+                    f"Unable to update stack component (type: {component.type}) "
+                    f"with name '{component.name}': No existing stack component "
+                    f"found with this name."
+                )
+            updated_component = session.exec(
+                select(ZenStackComponent)
+                .where(ZenStackComponent.component_type == component.type)
+                .where(ZenStackComponent.name == component.name)
+            ).first()
+            if updated_component:
+                updated_component.configuration = component.config
+
+            session.add(updated_component)
+            session.commit()
+            logger.info(
+                "Updated stack component with type '%s' and name '%s'.",
+                component.type,
+                component.name,
+            )
+
+    def rename_stack_component(
+        self,
+        old_name: str,
+        component: StackComponentWrapper,
+    ) -> None:
+        """Rename a stack component."""
+        with Session(self.engine) as session:
+            current_component = session.exec(
+                select(ZenStackComponent)
+                .where(ZenStackComponent.component_type == component.type)
+                .where(ZenStackComponent.name == old_name)
+            ).one_or_none()
+            if not current_component:
+                raise StackComponentExistsError(
+                    f"Unable to rename stack component (type: {component.type}) "
+                    f"with name '{component.name}': No existing stack component "
+                    f"found with this name."
+                )
+
+            # update the component itself
+            current_component.name = component.name
+            session.add(current_component)
```

Comment:
Weirdly, SQLModel doesn't have an `update` method. They use `add` for updating. See [here for the docs](https://sqlmodel.tiangolo.com/tutorial/update/#add-the-hero-to-the-session), in case I'm missing something? I think I test to make sure that we're not just creating two components, but 100% agreed that there is a lot of magic going on under the hood.